app/src/main/java/org/fossasia/openevent/general/speakers/SpeakerFragment.kt

```diff
@@ -0,0 +1,153 @@
+import org.fossasia.openevent.general.utils.Utils.setToolbar
+import org.fossasia.openevent.general.R
+import org.fossasia.openevent.general.utils.Utils.openUrl
+import org.fossasia.openevent.general.utils.Utils.requireDrawable
+import org.fossasia.openevent.general.utils.extensions.nonNull
+import org.fossasia.openevent.general.utils.stripHtml
+import org.jetbrains.anko.design.snackbar
+import org.koin.androidx.viewmodel.ext.android.viewModel
+
+class SpeakerFragment : Fragment() {
+    private lateinit var rootView: View
+    private val speakerViewModel by viewModel<SpeakerViewModel>()
+    private val safeArgs: SpeakerFragmentArgs by navArgs()
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        rootView = layoutInflater.inflate(R.layout.fragment_speaker, container, false)
+
+        setToolbar(activity)
+        setHasOptionsMenu(true)
+
+        speakerViewModel.error
+            .nonNull()
+            .observe(viewLifecycleOwner, Observer {
+                rootView.snackbar(it)
+            })
+
+        speakerViewModel.speaker
+            .nonNull()
+            .observe(viewLifecycleOwner, Observer {
+                initializeViews(it)
+            })
+
+        speakerViewModel.loadSpeaker(safeArgs.speakerId)
+
+        return rootView
+    }
+
+    private fun initializeViews(speaker: Speaker) {
+        when (speaker.name.isNullOrBlank()) {
+            true -> rootView.name.isVisible = false
+            false -> {
+                rootView.name.text = speaker.name
+                setToolbar(activity, speaker.name)
+            }
+        }
+
+        when (speaker.position.isNullOrBlank()) {
+            true -> rootView.position.isVisible = false
+            false -> rootView.position.text = speaker.position
+        }
+
+        when (speaker.organisation.isNullOrBlank()) {
+            true -> rootView.organization.isVisible = false
+            false -> rootView.organization.text = speaker.organisation
+        }
+
+        if (speaker.longBiography.isNullOrBlank()) {
+            if (speaker.shortBiography.isNullOrBlank())
+                rootView.aboutSpeakerContainer.isVisible = false
+            else rootView.bio.text = speaker.shortBiography.stripHtml()
+        } else rootView.bio.text = speaker.longBiography.stripHtml()
+
+        if (speaker.email.isNullOrBlank() && speaker.mobile.isNullOrBlank() &&
+            speaker.city.isNullOrBlank() && speaker.country.isNullOrBlank() &&
+            speaker.website.isNullOrBlank() && speaker.twitter.isNullOrBlank() &&
+            speaker.facebook.isNullOrBlank() && speaker.linkedin.isNullOrBlank() &&
+            speaker.github.isNullOrBlank()) rootView.speakerContactsContainer.isVisible = false
+        else {
+            when (speaker.email.isNullOrBlank()) {
+                true -> rootView.email.isVisible = false
+                false -> rootView.email.text = "Email: " + speaker.email
```

Comment:
Use `$` for string concatenation. Elsewhere too

Reply:
okay

Reply:
Done